### PR TITLE
Measure model memory usage

### PR DESCRIPTION
**Pull request by mgoin**
There is already a measure for kv cache blocks memory usage, indirectly through how many blocks were allocated, but no direct measure of how much memory the model weights are using. This PR tries to add that by wrapping the model loading with `torch.cuda.max_memory_allocated()` calls. I'm not sure how this will work with non-Nvidia devices, so happy to disable this in that case

Exposes a new `ModelRunner.model_memory_usage` member variable

Example code:
```python
from vllm import LLM
LLM("facebook/opt-125m")
```

Output:
```
INFO 02-29 19:21:40 llm_engine.py:79] Initializing an LLM engine with config: model='facebook/opt-125m', tokenizer='facebook/opt-125m', tokenizer_mode=auto, revision=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.float16, max_seq_len=2048, download_dir=None, load_format=auto, tensor_parallel_size=1, disable_custom_all_reduce=False, quantization=None, sparsity=None, enforce_eager=False, kv_cache_dtype=auto, device_config=cuda, seed=0)
INFO 02-29 19:21:44 weight_utils.py:176] Using model weights format ['*.bin']
>> INFO 02-29 19:21:45 model_runner.py:90] Loading model weights took 244.61 MB
INFO 02-29 19:21:45 llm_engine.py:338] # GPU blocks: 76240, # CPU blocks: 7281
```

### CODE REVIEW

**Comment by mgoin** [comment]
Hey @simon-mo , what do you think about this?

File: vllm/utils.py [hunk 1]
```diff
@@ -309,3 +310,27 @@ def create_kv_caches_with_random(
...
+
+class measure_cuda_memory:
+
+    def __init__(self, device=None):
+        self.device = device
+
+    def current_memory_usage(self) -> float:
+        # Return the memory usage in MB.
+        torch.cuda.reset_peak_memory_stats(self.device)
+        mem = torch.cuda.max_memory_allocated(self.device) / float(2**20)
```
Comment (zhuohan123) [inline vllm/utils.py:323]: Let's return bytes instead of MB here?

File: vllm/worker/model_runner.py [hunk 2]
```diff
@@ -85,11 +85,16 @@ def __init__(
...
+        with measure_cuda_memory() as m:
+            self.model = get_model(self.model_config,
+                                   self.device_config,
+                                   lora_config=self.lora_config,
+                                   parallel_config=self.parallel_config,
+                                   scheduler_config=self.scheduler_config)
+
+        self.model_memory_usage = m.consumed_memory
+        logger.info(
+            f"Loading model weights took {self.model_memory_usage:.2f} MB")
```
Comment (zhuohan123) [inline vllm/worker/model_runner.py:97]: Let's print with GB here?

**Review by zhuohan123 — APPROVED** [review]
LGTM! Left some small comments

**Comment by esmeetu** [comment]
Is that right with `tensor_parallel_size > 1`?

File: vllm/worker/model_runner.py (hunk 2 above)
Comment (mgoin) [reply vllm/worker/model_runner.py:97]: Updated

**Pull request merged**